### ISSUE TITLE
NoSuchMethodError - IndyBootstrap - Advice threw an exception - Lettuce

### ISSUE THREAD

**Issue by thaarbach** [elastic/apm-agent-java#3872]
**NoSuchMethodError**

APM Agent Version: 1.5.2
Lettuce Version: 6.5.0.RELEASE

```
2024-11-15 11:17:50,257 [http-nio-8080-exec-10] ERROR co.elastic.apm.agent.bci.IndyBootstrap - Advice threw an exception, this should never happen!
java.lang.NoSuchMethodError: 'java.lang.String io.lettuce.core.protocol.ProtocolKeyword.name()'
	at co.elastic.apm.agent.lettuce.Lettuce5StartSpanInstrumentation$AdviceClass.beforeDispatch(Lettuce5StartSpanInstrumentation.java:77) ~[elastic-apm-agent-4813494d137e1631bba301d5acab6e7b-ef6c8f75bd6181e717cdd17286444158.jar:1.52.0]
	at io.lettuce.core.RedisChannelHandler.dispatch(RedisChannelHandler.java:186) ~[lettuce-core-6.5.0.RELEASE.jar:6.5.0.RELEASE/7f455ec]
	at io.lettuce.core.StatefulRedisConnectionImpl.dispatch(StatefulRedisConnectionImpl.java:172) ~[lettuce-core-6.5.0.RELEASE.jar:6.5.0.RELEASE/7f455ec]
	at io.lettuce.core.AbstractRedisAsyncCommands.dispatch(AbstractRedisAsyncCommands.java:730) ~[lettuce-core-6.5.0.RELEASE.jar:6.5.0.RELEASE/7f455ec]
	at io.lettuce.core.AbstractRedisAsyncCommands.clusterInfo(AbstractRedisAsyncCommands.java:479) ~[lettuce-core-6.5.0.RELEASE.jar:6.5.0.RELEASE/7f455ec]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77) ~[?:?]
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
	at java.lang.reflect.Method.invoke(Method.java:569) ~[?:?]
	at io.lettuce.core.FutureSyncInvocationHandler.handleInvocation(FutureSyncInvocationHandler.java:67) ~[lettuce-core-6.5.0.RELEASE.jar:6.5.0.RELEASE/7f455ec]
	at io.lettuce.core.internal.AbstractInvocationHandler.invoke(AbstractInvocationHandler.java:84) ~[lettuce-core-6.5.0.RELEASE.jar:6.5.0.RELEASE/7f455ec]
	at jdk.proxy3.$Proxy281.clusterInfo(Unknown Source) ~[?:?]
```

In Version 6.5.0 of Lettuce the interface ProtocolKeyword has changed to 
```
package io.lettuce.core.protocol;

/**
 * Interface for protocol keywords providing an encoded representation.
 *
 * @author Mark Paluch
 */
public interface ProtocolKeyword {

    /**
     *
     * @return byte[] encoded representation.
     */
    byte[] getBytes();

    /**
     *
     * @return name of the command.
     */
    String toString();

}
```
https://github.com/redis/lettuce/blob/7f455ec0a69e0796d2bfcffed92103e634ac1fd0/src/main/java/io/lettuce/core/protocol/ProtocolKeyword.java#L20

Version 6.4.1 
```
package io.lettuce.core.protocol;

/**
 * Interface for protocol keywords providing an encoded representation.
 *
 * @author Mark Paluch
 */
public interface ProtocolKeyword {

    /**
     *
     * @return byte[] encoded representation.
     */
    byte[] getBytes();

    /**
     *
     * @return name of the command.
     */
    String name();

}
```

https://github.com/redis/lettuce/blob/08096cf86e24963c913562a0e0b725b11a8e08ff/src/main/java/io/lettuce/core/protocol/ProtocolKeyword.java#L20

**Comment by thaarbach** [comment]
Issue closed because issue seems to fixed in coming apm-agent-java

https://github.com/elastic/apm-agent-java/blob/7faedf22ce5304e57994b490dcd995b36478caca/apm-agent-plugins/apm-redis-plugin/apm-lettuce-plugin/src/main/java/co/elastic/apm/agent/lettuce/Lettuce5StartSpanInstrumentation.java#L77

**Comment by JonasKunz** [comment]
Fixed via #3826, should be available with 1.52.1
